src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonConcurrentQueueOfTConverter.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class JsonConcurrentQueueOfTConverter<TCollection, TElement> : JsonIEnumerableDefaultConverter<TCollection, TElement>
+        where TCollection : ConcurrentQueue<TElement>
+    {
+        protected override void Add(TElement value, ref ReadStack state)
+        {
+            ((TCollection)state.Current.ReturnValue!).Enqueue(value);
+        }
+
+        protected override void CreateCollection(ref ReadStack state, JsonSerializerOptions options)
+        {
+            if (state.Current.JsonClassInfo.CreateObject == null)
+            {
+                ThrowHelper.ThrowNotSupportedException_SerializationNotSupportedCollection(state.Current.JsonClassInfo.Type);
+            }
+
+            state.Current.ReturnValue = state.Current.JsonClassInfo.CreateObject();
+        }
+
+        protected override bool OnWriteResume(Utf8JsonWriter writer, TCollection value, JsonSerializerOptions options, ref WriteStack state)
+        {
+            IEnumerator<TElement> enumerator;
+            if (state.Current.CollectionEnumerator == null)
+            {
+                enumerator = value.GetEnumerator();
+                if (!enumerator.MoveNext())
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                enumerator = (IEnumerator<TElement>)state.Current.CollectionEnumerator;
+            }
+
+            JsonConverter<TElement> converter = GetElementConverter(ref state);
+            do
+            {
+                if (ShouldFlush(writer, ref state))
+                {
+                    state.Current.CollectionEnumerator = enumerator;
+                    return false;
+                }
+
+                TElement element = enumerator.Current;
+                if (!converter.TryWrite(writer, element, options, ref state))
+                {
+                    state.Current.CollectionEnumerator = enumerator;
+                    return false;
+                }
+
+                state.Current.EndElement();
+            } while (enumerator.MoveNext());
+
+            return true;
+        }
+
+        internal override Type RuntimeType => typeof(Queue<TElement>);
```

Comment:
Why this converter uses hardcoded `Queue<T>` and not `ConcurrentQueue<T>` or `TypeToConvert`?

cc @layomia

Reply:
Good catch. The override isn't needed here.

Reply:
The override is not needed; removing.